### ISSUE TITLE
Add "dummy_data" module

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#1768]
Currently, if we want to make docstring examples, we often have to construct signals by generating NumPy arrays with some kind of data.

This is fine for simple functionality, but for more advanced processing functions, like my [`set_coreloss_edge_onset`](https://github.com/hyperspy/hyperspy/pull/1271/files#diff-6a2092a2cb15bef60d59bb3ee3c49b36R1017) method, this would require several lines of code to simply make a minimum working example.

It would be much better if we had a `hs.dummy_data` module, which could contain functions which generate artificial data to be used in the user guide or docstring. This would make it a lot easier to write instructive docstrings, and also allow for the docstrings to be run as unit tests. Since currently, very many of the docstrings won't run on their own as unit tests.

An example of this for my `set_coreloss_edge_onset` function could be:

```python
import numpy as np
from hyperspy import components1d
from hyperspy.signals import EELSSpectrum

def get_artifical_core_loss_eel_model():
    x = np.arange(400, 800, 1)
    arctan = components1d.Arctan(A=1, k=0.2, x0=688)
    arctan.minimum_at_zero = True
    mn_l3_g = components1d.Gaussian(A=100, centre=695, sigma=4)
    mn_l2_g = components1d.Gaussian(A=20, centre=720, sigma=4)

    data = arctan.function(x)
    data += mn_l3_g.function(x)
    data += mn_l2_g.function(x)
    data += np.random.random(size=len(x))*0.7

    s = EELSSpectrum(data)
    s.axes_manager[0].offset = x[0]
    s.metadata.General.title = 'Artifical core loss EEL spectrum'
    s.axes_manager[0].name = 'Electron energy loss'
    s.axes_manager[0].units = 'eV'
    s.set_microscope_parameters(
            beam_energy=200, convergence_angle=26, collection_angle=20)
    s.add_elements(('Mn', ))
    m = s.create_model(auto_background=False, GOS='hydrogenic')
    m.fit()
    return m
```

This function gives this model:

![hyperspy_dummy_data](https://user-images.githubusercontent.com/1690979/31861082-4bcf83a2-b71e-11e7-9de0-a0c088eb1871.png)

This could make the docstring for `set_coreloss_edge_onset`:

```python
m = hs.dummy_data.get_artifical_core_loss_eel_model()
m.plot()
m.set_coreloss_edge_onset(m.components.Mn_L3, signal_range=(670, 700))
```

We've done this in the development version of Atomap: [`atomap.dummy_data`](https://gitlab.com/atomap/atomap/blob/master/atomap/dummy_data.py), which has made it a great deal easier to make concise docstring, and user guides, which also can be tested using doctest.

An example of this in Atomap:
```python
>>> import atomap.api as am
>>> sublattice = am.dummy_data.get_simple_cubic_sublattice()
>>> s = sublattice.get_property_map(
...                    sublattice.x_position,
...                    sublattice.y_position,
...                    sublattice.ellipticity)
>>> s.plot()
```

**Comment by ericpre** [comment]
There is already the `hyperspy.datasets` module containing example signals and also giving access to the [EELS database](https://eelsdb.eu).
Maybe you could add your function to `hyperspy.datasets` in a `artificial_data` module (or something similar)?

**Comment by magnunor** [comment]
Good point! I'd forgotten about the `hyperspy.datasets` module.

`artificial_data` sounds like a good name. I'll make a pull request with some simple artificial signals, including the one above.

**Comment by ericpre** [comment]
Implemented in #1789.
